ystia/yorc
# Feature/ansible task output asap

# Pull Request description

## Description of the change

Instead of displaying Ansible playbook logs at the end of the playbook,
providing each task log once the task has finished without having to
wait for the playbook end.

### How I did it

Configuring Ansible to provide yaml output, instead of json output that
provides logs only at the end of the playbook.

The yaml output feature is provided since Ansible 2.5, so an upgrade of
the Ansible version used by Yorc is necessary. Upgrading Ansible to
version 2.6.3.

Parsing yaml output and displaying it as it is in logs for ansible
playbook. While when running scripts, providing only the script output
as the fact that the script is run in an Ansible playbook is an
orchestrator implementation detail.

For example it gives this logs for an Ansible playbook task execution :
```
[2018-08-23T17:30:14.44177507Z][INFO][test-Environment][install][c5ae0b5b-91a2-4517-b709-a990a5854255][LongStartupAnsible][0][standard][start][]Ansible task output:
PLAY [Print number of iterations in first task, iterate in second task] TASK [First task]
changed: [192.168.2.11] => changed=true
   cmd: ~/create.sh
   delta: '0:00:00.005701'
   end: '2018-08-23 17:29:49.794012'
   invocation:
     module_args:
       _raw_params: ~/create.sh
       _uses_shell: true
       argv: null
       chdir: null
       creates: null
       executable: null
       removes: null
       stdin: null
       warn: true
   rc: 0
   start: '2018-08-23 17:29:49.788311'
   stderr: |-
     Writing to stderr line 1
     Writing to stderr line 2
   stderr_lines:
   - Writing to stderr line 1
   - Writing to stderr line 2
   stdout: Created component to iterate 12 times, with a pause of 2
seconds after each iteration
   stdout_lines: <omitted>
```

and this log for a script execution :
```
2018-08-23T17:30:16.079046877Z][WARN][test-Environment][install][c5ae0b5b-91a2-4517-b709-a990a5854255][LongStartupBash][0][standard][start][]Node "LongStartupBash", host "192.168.2.11", stderr:
Error message 1
Error message 2
[2018-08-23T17:30:16.083279577Z][INFO][test-Environment][install][c5ae0b5b-91a2-4517-b709-a990a5854255][LongStartupBash][0][standard][start][]Node "LongStartupBash", host "192.168.2.11", stdout:
Long startup component iteration 1
Long startup component iteration 2
Long startup component iteration 3
Long startup component iteration 4
```
### How to verify it

Use the yorc docker image generated for this Pull Request, deploy an
application and check logs.


## Applicable Issues

#146
